sci-geosciences/josm-bin/josm-bin-17428.ebuild

```diff
@@ -0,0 +1,42 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit java-utils-2 desktop
+
+DESCRIPTION="Java-based editor for the OpenStreetMap project"
+HOMEPAGE="https://josm.openstreetmap.de/"
+
+SRC_URI="https://josm.openstreetmap.de/download/josm-snapshot-${PV}.jar"
+
+LICENSE="GPL-2"
+SLOT="0"
+
+KEYWORDS="~amd64 ~x86"
+
+DEPEND=""
+RDEPEND=">=virtual/jre-1.8"
+
+IUSE=""
+
+S=${WORKDIR}
+
+src_unpack() {
+	default
+	cp "${DISTDIR}/${A}" "${S}/${PN}.jar"
```

Comment:
`default` is going to unpack the jar archive, which is not necessary here. Instead, you can just skip the `src_unpack` phase completely, and directly do `java-pkg_newjar "${DISTDIR}/josm-snapshot-${PV}.jar" "${PN}.jar"` in `src_install()`.

Reply:
moved to java-pkg_newjar, can not skip unpack since this gives us the icons